share/dkg/dkg.go

```diff
@@ -0,0 +1,654 @@
+// Package dkg implements the protocol described in
+// "Secure Distributed Key Generation for Discrete-Log
+// Based Cryptosystems".
```

Comment:
Note that this paper seems to be only available behind the Springer paywall, unfortunately. I am not sure how much benefit it adds if we include a url to Springer? We should definitely add the list of authors though. Maybe something along the lines of

```
Secure Distributed Key Generation for Discrete-Log Based Cryptosystems" by R. Gennaro, S. Jarecki, H. Krawczyk, and T. Rabin
```